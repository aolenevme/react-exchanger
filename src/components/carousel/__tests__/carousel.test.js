// eslint-disable-next-line import/no-namespace
import * as React from "react";
import renderer from "react-test-renderer";

import Carousel from "../carousel.js";

describe("<Carousel />", () => {
    describe("rendering", () => {
        it("is rendered correctly", () => {
            expect(renderer
                .create(<Carousel />)
                .toJSON()).toMatchSnapshot();
        });
    });

    describe("scrolling", () => {
        it("scrolls to the right", () => {
            const currentPocketIdx = 1;
            const startX = 1;
            const endX = 1;
            const scrollIntoView = jest.fn();

            // eslint-disable-next-line fp/no-mutation,no-import-assign
            React.useState = jest
                .fn()
                .mockReturnValue("default")
                .mockReturnValueOnce([currentPocketIdx, jest.fn(() => currentPocketIdx + 1)])
                .mockReturnValueOnce([startX, jest.fn()])
                .mockReturnValueOnce([endX, jest.fn()]);

            // eslint-disable-next-line fp/no-mutation,no-import-assign
            React.useRef = jest.fn(() => [{current: {children: [{}, {}, {scrollIntoView}]}}]);

            expect(scrollIntoView).toHaveBeenCalledWith({block: "start", inline: "nearest", behavior: "smooth"});
        });

        // 0. Probably, have to pass into pocketsInfo

        // 0*. Probably, has to check pocketsInfo for valid fields

        // 1. Set currentPocketIdx = 1 -> endX = 2 -> startX = 1, check that carouselElement.current.children[currentPocketIdx + 1].scrollIntoView
        // === jest.fn was called with {block: "start", inline: "nearest", behavior: "smooth"}. And test Dots via snapshot tests.

        // 2. Set currentPocketIdx = 1 -> endX = 2 -> startX = 3, check that carouselElement.current.children[currentPocketIdx - 1].scrollIntoView
        // === jest.fn was called with {block: "start", inline: "nearest", behavior: "smooth"}. And test Dots via snapshot tests.

        // 3. Set currentPocketIdx = 2 -> endX = 2 -> startX = 1, check that carouselElement.current.children[currentPocketIdx + 1].scrollIntoView
        // === jest.fn WAS NOT called with {block: "start", inline: "nearest", behavior: "smooth"}. And test Dots via snapshot tests.

        // 4. Set currentPocketIdx = 0 -> endX = 2 -> startX = 1, check that carouselElement.current.children[currentPocketIdx + 1].scrollIntoView
        // === jest.fn WAS NOT called with {block: "start", inline: "nearest", behavior: "smooth"}. And test Dots via snapshot tests.
    });
});
