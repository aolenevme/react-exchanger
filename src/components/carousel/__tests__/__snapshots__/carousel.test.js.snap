// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> rendering is rendered correctly 1`] = `
.c8 {
  font-size: 0.8rem;
  color: #7199e6;
}

.c4 {
  font-size: 3rem;
  line-height: 3rem;
  color: #f3f6ff;
}

.c1 {
  padding: 2rem;
  width: 100%;
  box-sizing: border-box;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 0.3rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  overflow: hidden;
}

.c2 {
  -webkit-flex: 0 0 100%;
  -ms-flex: 0 0 100%;
  flex: 0 0 100%;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-bottom: 2rem;
}

.c10 {
  display: inline-block;
  margin: 0 0.142rem;
  width: 0.5rem;
  height: 0.5rem;
  border-radius: 50%;
  background-color: #f3f6ff;
  -webkit-transition: background-color 0.1s ease 0s;
  transition: background-color 0.1s ease 0s;
}

.c11 {
  display: inline-block;
  margin: 0 0.142rem;
  width: 0.5rem;
  height: 0.5rem;
  border-radius: 50%;
  background-color: #5591f7;
  -webkit-transition: background-color 0.1s ease 0s;
  transition: background-color 0.1s ease 0s;
}

.c5 {
  font-size: 3rem;
  line-height: 3rem;
  color: #f3f6ff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 0;
  min-width: 3rem;
  height: 3rem;
  border: 0 solid transparent;
  border-right: 1px solid transparent;
  background-color: transparent;
  cursor: pointer;
}

.c5:focus-within {
  -webkit-animation: caret-pulse 1.5s cubic-bezier(.215,.61,.355,1) forwards infinite;
  animation: caret-pulse 1.5s cubic-bezier(.215,.61,.355,1) forwards infinite;
}

.c6 {
  margin: 0;
  padding: 0;
  width: 0;
  height: 0;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c6::-webkit-outer-spin-button,
.c6::-webkit-inner-spin-button {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c6:invalid {
  box-shadow: none;
}

<div>
  <div
    className="c0"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="c1 c2"
    >
      <div
        className="c3"
      >
        <span
          className="c4"
        >
          USD
        </span>
        <label
          className="c5"
        >
          -
          145.67
          <input
            className="c6"
            disabled={false}
            onInput={[Function]}
            type="number"
            value={145.67}
          />
        </label>
      </div>
      <div
        className="c7"
      >
        <span
          className="c8"
        >
          You have 58.33$
        </span>
        <span
          className="c8"
        >
          £1 = $1.45
        </span>
      </div>
    </div>
    <div
      className="c1 c2"
    >
      <div
        className="c3"
      >
        <span
          className="c4"
        >
          EUR
        </span>
        <label
          className="c5"
        >
          -
          145.67
          <input
            className="c6"
            disabled={false}
            onInput={[Function]}
            type="number"
            value={145.67}
          />
        </label>
      </div>
      <div
        className="c7"
      >
        <span
          className="c8"
        >
          You have 58.33$
        </span>
        <span
          className="c8"
        >
          £1 = $1.45
        </span>
      </div>
    </div>
    <div
      className="c1 c2"
    >
      <div
        className="c3"
      >
        <span
          className="c4"
        >
          GBP
        </span>
        <label
          className="c5"
        >
          -
          145.67
          <input
            className="c6"
            disabled={false}
            onInput={[Function]}
            type="number"
            value={145.67}
          />
        </label>
      </div>
      <div
        className="c7"
      >
        <span
          className="c8"
        >
          You have 58.33$
        </span>
        <span
          className="c8"
        >
          £1 = $1.45
        </span>
      </div>
    </div>
  </div>
  <div
    className="c9"
  >
    <span
      className="c10"
    />
    <span
      className="c11"
    />
    <span
      className="c11"
    />
  </div>
</div>
`;

exports[`<Carousel /> rendering is rendered correctly 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  overflow: hidden;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-bottom: 2rem;
}

<div>
  <div
    className="c0"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  />
  <div
    className="c1"
  />
</div>
`;
